### PR TITLE
[PC-6484] Geolocation request spike

### PR DESCRIPTION
**Pull request by justinemignot**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-6484

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [x] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [x] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

Cf vidéo "Comportement-implémenté-android" dans le ticket 

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/search/pages/LocationFilter.tsx [hunk 1]
```diff
@@ -28,9 +30,9 @@ export const LocationFilter: React.FC = () => {
     navigate('LocationPicker')
   }
 
-  const onPressAroundMe = () => {
+  const onPressAroundMe = async () => {
```
Comment (antoinewg) [inline src/features/search/pages/LocationFilter.tsx:33]: besoin du async ?

File: src/libs/geolocation/useRequestGeolocPermission.android.ts [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
-      PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
-      PermissionsAndroid.PERMISSIONS.ACCESS_COARSE_LOCATION,
-    ]).then((permissions) => {
-      if (
-        permissions[PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION] === 'granted' ||
-        permissions[PermissionsAndroid.PERMISSIONS.ACCESS_COARSE_LOCATION] === 'granted'
-      ) {
-        setPermissionGranted(true)
-      }
+export const useRequestGeolocPermission = (setPermissionGranted: (granted: boolean) => void) => {
```
Comment (antoinewg) [inline src/libs/geolocation/useRequestGeolocPermission.android.ts:3]: enlenver le `use` vu que c'est plus un hook.

File: src/libs/geolocation/useRequestGeolocPermission.d.ts [hunk 3]
```diff
@@ -1 +1,5 @@
-export const useRequestGeolocPermission: () => boolean
+export const useRequestGeolocPermission: (
+  setPermissionGranted: (granted: boolean) => void
```
Comment (antoinewg) [inline src/libs/geolocation/useRequestGeolocPermission.d.ts:2]: je verrais bien `onGranted` comme argument à cette fonction. Parce que `setPermissionGranted`, ça fait bien setState, je sais pas si c'est voulu

Comment (antoinewg) [reply src/libs/geolocation/useRequestGeolocPermission.d.ts:2]: c'est pas trop important ça

File: src/libs/geolocation/useRequestGeolocPermission.ios.ts [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
 
-  return permissionGranted
+  return {
+    requestPermissionRoutine,
+  }
 }
+
+export const requestGeolocPermission = (): Promise<boolean> =>
+  Geolocation.requestAuthorization('whenInUse').then((value) => {
+    if (value === 'granted') return true
```
Comment (antoinewg) [inline src/libs/geolocation/useRequestGeolocPermission.ios.ts:17]: return `value === "granted"`

File: src/libs/geolocation/GeolocationWrapper.tsx [hunk 5]
```diff
@@ -7,27 +7,40 @@ import { getPosition } from './getPosition'
...
+  setPermissionGranted: () => undefined,
+})
 
 export const GeolocationWrapper = ({ children }: { children: Element }) => {
-  const [position, setInitialPosition] = useState<GeoCoordinates | null>(null)
-  const permissionGranted = useRequestGeolocPermission()
+  const [position, setPosition] = useState<GeoCoordinates | null>(null)
+  const [permissionGranted, setPermissionGranted] = useState<boolean>(false)
+  const { requestPermissionRoutine } = useRequestGeolocPermission(setPermissionGranted)
+  requestPermissionRoutine()
```
Comment (antoinewg) [inline src/libs/geolocation/GeolocationWrapper.tsx:26]: pk on appelle ça `Routine` ?

File: src/features/search/pages/LocationFilter.tsx (hunk 1 above)
Comment (justinemignot) [reply src/features/search/pages/LocationFilter.tsx:33]: indeed

File: src/libs/geolocation/useRequestGeolocPermission.android.ts (hunk 2 above)
Comment (justinemignot) [reply src/libs/geolocation/useRequestGeolocPermission.android.ts:3]: 👍

File: src/libs/geolocation/useRequestGeolocPermission.d.ts (hunk 3 above)
Comment (justinemignot) [reply src/libs/geolocation/useRequestGeolocPermission.d.ts:2]: done

**Review by antoinewg — APPROVED** [review]

**Pull request merged**